### DISCUSSION TITLE
bash: ble/_ble_util_openat_fdlistay#push: No such file or directory

### DISCUSSION THREAD

**Discussion by jkemp814** [akinomyoga/ble.sh#376] [General]
More error messages while updating Oh-My-Bash.

04:07:27 james@Dell-7740 ~ → bash: ble/_ble_util_openat_fdlistay#push: No such file or directory
bash: ble/_ble_highlight_layer_listay#insert-at: No such file or directory
bash: ble/_ble_util_openat_fdlistay#push: No such file or directory

When using `upgrade_oh_my_bash` after that there is no error message.

**Comment by akinomyoga** [comment]
I guess it is related to `shopt -s nocasematch`.

**Reply by akinomyoga** [reply]
I try to reproduce the problem, but I cannot reproduce it.

The behavior you observe would be caused by calling `ble/array#insert-before` or `ble/array#insert-after` in an environment where `shopt -s nocasematch` is set. In fact, OMB sets `shopt -s nocasematch`. However, even if OMB sets it, `ble/array#insert-before` or `ble/array#insert-after` is never called in the context with `shopt -s nocasematch` because most of the code of ble.sh are processed in a different context.

Could you provide the setup of your `.bashrc` and `.blerc`?

**Reply by akinomyoga** [reply]
I could reproduce a similar error message by putting some of the configurations in `.bashrc` instead of `.blerc`. One possibility I suspect is that you include configurations for ble.sh in `~/.bashrc` or some files directly sourced from `~/.bashrc`. If that is the case, could you put all the ble.sh-related settings in `~/.blerc` or in `~/.config/blesh/init.sh`?

**Comment by jkemp814** [comment]
I have hardly any config in my bashrc. I use it for the prompt ('font' theme for git printouts) But I realize there is a lot going on in the background with OMB.

One moment, please.

**Reply by akinomyoga** [reply]
Thanks. I tried your settings, but I still cannot reproduce the problem.

- **Q1**: Does the problem reproduce with the minimal setup? I.e. could you disable `fzf`, `zoxide`, and `cargo` settings and all the extra modules of Oh My Bash to see if the problem persists? I mean the testing with the changes like the following (where the red parts are the removed lines and the green parts are the added lines). After testing, you can revert the changes.

```diff
diff --git a/.bashrc b/.bashrc
index c6d1f18..650cfb3 100644
--- a/.bashrc
+++ b/.bashrc
@@ -16,22 +16,22 @@ OSH_THEME="font"
 export UPDATE_OSH_DAYS=13

 completions=(
-  defaults
-  git
-  ssh
+  # defaults
+  # git
+  # ssh
 )

 aliases=(
-  general
+  # general
 )

 plugins=(
-  bashmarks
-  git
-  zoxide
+  # bashmarks
+  # git
+  # zoxide
 )

 source "$OSH"/oh-my-bash.sh

 [[ ${BLE_VERSION-} ]] && ble-attach
-[ -f ~/.fzf.bash ] && source ~/.fzf.bash
+# [ -f ~/.fzf.bash ] && source ~/.fzf.bash
diff --git a/.profile b/.profile
index a383d9d..e65f585 100644
--- a/.profile
+++ b/.profile
@@ -5,7 +5,7 @@ if ! [[ "$PATH" =~ "$HOME/.local/bin:$HOME/bin:" ]]; then
 fi
 export PATH

-. "$HOME/.cargo/env"
-eval "$(zoxide init bash)"
-[ -f ~/.fzf.bash ] && source ~/.fzf.bash
-ANSIBLE_HOME=~/.ansible
+# . "$HOME/.cargo/env"
+# eval "$(zoxide init bash)"
+# [ -f ~/.fzf.bash ] && source ~/.fzf.bash
+# ANSIBLE_HOME=~/.ansible
```

**Comment by jkemp814** [comment]
.blerc (I removed everything that was commented out.

```bash
bleopt input_encoding=UTF-8

bleopt pager=less

bleopt editor=vi

bleopt vbell_default_message=' Wuff, -- Wuff!! '
bleopt vbell_duration=2000
bleopt vbell_align=right

bleopt edit_abell=1
bleopt edit_vbell=

bleopt history_lazyload=1

bleopt indent_offset=4
bleopt indent_tabs=1

bleopt edit_line_type=graphical

bleopt info_display=top

#bleopt prompt_ps1_final=
bleopt prompt_ps1_transient=trim

bleopt prompt_rps1='\w'
bleopt prompt_rps1_final=
bleopt prompt_rps1_transient=''

#function ble/prompt/backslash:custom/memoryTime {
#  ble/prompt/unit/add-hash '$SECONDS'
#  ble/prompt/process-prompt-string "💾 $(free -h | grep Mem | awk '{printf $3 " / " $2}')\r⏰ \t"
#}
#bleopt prompt_status_line=" 🐧 $(uname -o) $(uname -r) $(uname -m)\r📟$(dmesg | grep 'Machine model' | awk -F': ' '{print $2}')\r\q{custom/memoryTime}"
#bleopt prompt_status_align=$'justify=\r'
#bleopt exec_elapsed_mark=$' \\e[94m[ ↪️ [%s 📈 %s%%]\e[m'
#bleopt exec_elapsed_enabled='usr+sys>=10'
#bleopt exec_errexit_mark=$' \e[91m[⚡️RC ➡️ %d]\e[m'
#bleopt prompt_ruler=$'\e[94m-'

bleopt prompt_eol_mark=$'\e[94m[ble: EOF]\e[m'

bleopt emoji_width=2

bleopt emoji_version=13.1

bleopt emoji_opts=ri

bleopt grapheme_cluster=extended

bleopt canvas_winch_action=redraw-here

bleopt decode_error_char_vbell=1
bleopt decode_error_cseq_vbell=1
bleopt decode_error_kseq_vbell=1

function blerc/emacs-load-hook {
  return 0
}
blehook/eval-after-load keymap_emacs blerc/emacs-load-hook

function blerc/vim-load-hook {
  ((_ble_bash>=40300)) && builtin bind 'set keyseq-timeout 1'

}
blehook/eval-after-load keymap_vi blerc/vim-load-hook
```

**Comment by jkemp814** [comment]
.bashrc (removed everything that was commented out)


```bash
# Add this lines at the top of .bashrc:
[[ $- == *i* ]] && source "$HOME/.local/share/blesh/ble.sh" --noattach

# Enable the subsequent settings only in interactive sessions
case $- in
  *i*) ;;
    *) return;;
esac

# Path to your oh-my-bash installation.
export OSH='/var/home/james/.oh-my-bash'

# Set name of the theme to load. Optionally, if you set this to "random"
# it'll load a random theme each time that oh-my-bash is loaded.
OSH_THEME="font"

# Uncomment the following line to change how often to auto-update (in days).
export UPDATE_OSH_DAYS=13

# Which completions would you like to load? (completions can be found in ~/.oh-my-bash/completions/*)
# Custom completions may be added to ~/.oh-my-bash/custom/completions/
# Example format: completions=(ssh git bundler gem pip pip3)
# Add wisely, as too many completions slow down shell startup.
 completions=(
     defaults
     git
     ssh
)
 
# Which aliases would you like to load? (aliases can be found in ~/.oh-my-bash/aliases/*)
# Custom aliases may be added to ~/.oh-my-bash/custom/aliases/
# Example format: aliases=(vagrant composer git-avh)
# Add wisely, as too many aliases slow down shell startup.
aliases=(
    general
)

# Which plugins would you like to load? (plugins can be found in ~/.oh-my-bash/plugins/*)
# Custom plugins may be added to ~/.oh-my-bash/custom/plugins/
# Example format: plugins=(rails git textmate ruby lighthouse)
# Add wisely, as too many plugins slow down shell startup.
plugins=(
    bashmarks
    git
    zoxide
)

source "$OSH"/oh-my-bash.sh

# Add this line at the end of .bashrc:
[[ ${BLE_VERSION-} ]] && ble-attach
[ -f ~/.fzf.bash ] && source ~/.fzf.bash
```

**Comment by jkemp814** [comment]
.bash_profile file.

```bash
# .bash_profile -*- mode: sh -*-

# Load login settings and environment variables
if [[ -f ~/.profile ]]; then
  source ~/.profile
fi

# Load interactive settings
if [[ -f ~/.bashrc ]]; then
  source ~/.bashrc
fi
```

**Comment by jkemp814** [comment]
.profile file.

```bash
if ! [[ "$PATH" =~ "$HOME/.local/bin:$HOME/bin:" ]]; then
    PATH="$HOME/.local/bin:$HOME/bin:$PATH"
fi
export PATH

. "$HOME/.cargo/env"
eval "$(zoxide init bash)"
[ -f ~/.fzf.bash ] && source ~/.fzf.bash
ANSIBLE_HOME=~/.ansible
```

**Comment by jkemp814** [comment]
I comment out everything but:

.bashrc

export OSH='/var/home/james/.oh-my-bash'
export UPDATE_OSH_DAYS=13
source "$OSH"/oh-my-bash.sh

and everything in my .profile except the PATH statements.

I'm confused about those characters '@@ -16,22 +16,22 @@' and  '@@ -5,7 +5,7 @@'? Those are GitHub artifacts?

I do not receive the strange printouts when I manually run `upgrade_oh_my_bash`, but the cursor hangs, about 15 spaces to the right, and just blinks. I will leave everything commented out and set the upgrade to one day and see what happens when it runs on it own.

**Reply by akinomyoga** [reply]
Thanks for checking.

> I'm confused about those characters '@@ -16,22 +16,22 @@' and '@@ -5,7 +5,7 @@'? Those are GitHub artifacts?

These are a part of [the unified diff format](https://en.wikipedia.org/wiki/Diff#Unified_format), which is a way to represent edits of a file. These lines show the line numbers of the changed section in the old and new files. If you do not know it, you can just ignore it. It's a kind of code comment for "diff".

> but the cursor hangs, about 15 spaces to the right, and just blinks.

This one is reproduced in my environment (although the error messages are still not reproducible). This seems to be caused when `ble-attach` is executed in `~/.bashrc`. I'll look at it later.

**Comment by jkemp814** [comment]
No it did not give me the printouts, and did not cause the cursor to hang.

However, it does hang after I run the upgrade manually.

**Comment by jkemp814** [comment]
So, I think your suspicions were correct, it has something to do with one of those packages. I will enable them one at a time and check to see when it occurs.

**Reply by akinomyoga** [reply]
> it has something to do with one of those packages. I will enable them one at a time and check to see when it occurs.

Thanks. That is what I planned to ask you next when the error messages disappear. Could you try that?

**Comment by jkemp814** [comment]
Even when I `source ~/.bash_profile` the cursor hangs. I have to press enter in order to get back to the prompt.
